### PR TITLE
Bug Fixes, and updating gems and code

### PR DESCRIPTION
**Pull request by dartion5**

### CODE REVIEW

File: db/migrate/20111222215513_devise_create_users.rb [hunk 1]
```diff
@@ -5,7 +5,7 @@ def change
       t.recoverable
       t.rememberable
       t.trackable
-t.lockable :lock_strategy => :failed_attempts, :unlock_strategy => :time
+t.lockable :lock_strategy => :failed_attempts, :unlock_strategy => :both
```
Comment (shaunmuscat) [inline db/migrate/20111222215513_devise_create_users.rb]: this change should be made in a new migration file.

File: db/migrate/20111222215513_devise_create_users.rb [hunk 2]
```diff
@@ -22,8 +22,8 @@ def change
 
     add_index :users, :email,                :unique => true
     add_index :users, :reset_password_token, :unique => true
+    add_index :users, :unlock_token,         :unique => true
```
Comment (shaunmuscat) [inline db/migrate/20111222215513_devise_create_users.rb]: this change should be made in a new migration file.

**Comment by shaunmuscat** [comment]
This pull request provides solutions for https://jira.intersect.org.au/browse/DEVSUPPORT-1737 & https://jira.intersect.org.au/browse/DEVSUPPORT-1738.

**Comment by dartion5** [comment]
DEVSUPPORT-1737
DEVSUPPORT-1738

**Pull request merged**